src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -85,14 +87,11 @@ public FileUploadSubmissionResource(SubmissionRepository submissionRepository, R
 
         final var exercise = fileUploadExerciseRepository.findOneByIdElseThrow(exerciseId);
         final User user = userRepository.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isAtLeastStudentForExercise(exercise, user)) {
-            return forbidden();
-        }
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.STUDENT, exercise, user);
 
-        // Make sure that the exercise exists
-        if (exercise == null) {
-            return ResponseEntity.badRequest()
-                    .headers(HeaderUtil.createFailureAlert(applicationName, true, "submission", "exerciseNotFound", "No exercise was found for the given ID.")).body(null);
+        if (fileUploadSubmission.getParticipation() != null && fileUploadSubmission.getParticipation().getExercise() != null
+                && !fileUploadSubmission.getParticipation().getExercise().getId().equals(exerciseId)) {
+            return badRequest("exerciseId", "400", "ExerciseId in Body doesnt match ExerciseId in path!");
```

Comment:
To improve readability, maybe this one needs a short comment above: "if there is a participation containing an exercise, make sure its it match the given path variable". Also, I would write " does not".